src/camera.c

```diff
@@ -126,7 +126,7 @@ query_permission_sync (Request *request)
 
       if (!xdp_dbus_impl_access_call_access_dialog_sync (access_impl,
                                                          request->id,
-                                                         app_id,
+                                                         info ? app_id : "",
```

Comment:
So checked what happens on uninstallation, and this indeed bypasses the check if one runs an sandboxed app, uninstalls it, then tries to get camera access.

I suggest something like

```
if (xdp_app_info_is_host (request->app_info))
  queried_app_id = info ? app_id : "";
else
  queried_app_id = app_id;
```
or something.